src/plugins/graphs/star/plugin.cpp

```diff
@@ -44,7 +44,7 @@ bool StarGraph::reset()
     // which will always start from 0 and end at size-1
     node(0).setCoords(radius, radius);
     if (m_edgeAttrsGen) {
-        auto soa = m_edgeAttrsGen->create(nNodes - 1);
+        auto soa = m_edgeAttrsGen->create(nNodes);
         for (int nodeId = 1; nodeId < nNodes; ++nodeId) {
             fixCoords(node(nodeId), radius, dTheta);
             addEdge(0, nodeId, new Attributes(soa.at(nodeId)));
```

Comment:
Can we initialize an experiment without nodes in Evoplex? How?
I guess that you are trying to call reset() programmatically from some new code in the GraphDesigner, right?

In any case, if nNodes is <2, the graph wouldn't have edges at all, then why would we need an instance of the `attrsGen` here? (note that the loop goes from 1 to n-1)

I suppose that the correct fix here would be to check
``` c++
if (nNodes == 0) {
    return true;
} else if (nNodes == 1) {
    node(0).setCoords(0, 0);
    return true;
}
```
after line 35

Reply:
@cardinot I didn't encounter the error when testing with 1 node. I noticed that I got a segmentation fault whenever I tried to generate a Star graph with ``m_edgeAttrsGen`` when adding the last node of the graph.

Correct me if I'm wrong, but this is what I think happens:

- ``soa`` is initialized with indices ranging from 0 to ``nNodes - 2``

- The loop iterates from 1 to ``nNodes - 1``

So when adding the last node, the program crashes when it gets to ``soa.at(nodeId)`` since ``nNodes - 1`` isn't a valid key of soa.

Reply:
ah, ok; that's correct
so, I suppose that the correct fix here would be to change L50 to:
``` c++
addEdge(0, nodeId, new Attributes(soa.at(nodeId - 1)));
```
do you agree?

Reply:
to improve readability, perhaps we could do something like:
``` c++
        auto setOfAttrs = m_edgeAttrsGen->create(nNodes - 1);
        int nodeId = 1;
        for (auto& attrs : setOfAttrs) {
            fixCoords(node(nodeId), radius, dTheta);
            addEdge(0, nodeId, new Attributes(attrs));
            ++nodeId;
        }
```
what do you think?

Reply:
Ah that looks better and less confusing!